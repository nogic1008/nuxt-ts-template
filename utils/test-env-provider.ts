/* eslint-disable no-process-env */
/** Provides process.env for Testing. */
export default class TestEnvProvider<T> {
  /** Stored process.env */
  private readonly storedEnv: NodeJS.ProcessEnv
  private readonly keys: Set<Extract<keyof T, string>>

  constructor(...keys: Extract<keyof T, string>[]) {
    this.storedEnv = { ...process.env }
    this.keys = new Set(keys)
  }

  initialize(obj?: Partial<T>) {
    const env = { ...this.storedEnv }
    this.keys.forEach((k) => delete env[k])
    process.env = { ...env, ...obj }
  }

  restore() {
    process.env = { ...this.storedEnv }
  }
}
