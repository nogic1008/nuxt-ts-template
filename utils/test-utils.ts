/** Generate [0-9a-z]{length} string. */
export const generateRandomString = (length: number) =>
  [...Array(Number.isInteger(length) ? length : 12)]
    .map(() => (~~(Math.random() * 36)).toString(36))
    .join('')
