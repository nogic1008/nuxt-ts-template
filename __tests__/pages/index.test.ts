import { describe, expect, test } from '@jest/globals'
import { mount } from '@vue/test-utils'

import { createI18n, createVue } from '~/__tests__/utils'
import Index from '~/pages/index.vue'

const localVue = createVue()

describe('pages/index.vue', () => {
  describe.each(['en', 'ja'])(
    '{ locale: %s } snapshot test',
    (locale: string) => {
      const i18n = createI18n(locale)
      test('renders correctly', () => {
        const stubs = ['card', 'counter']
        const wrapper = mount(Index, { localVue, stubs, i18n })
        expect(wrapper.element).toMatchSnapshot()
      })
    }
  )
})
