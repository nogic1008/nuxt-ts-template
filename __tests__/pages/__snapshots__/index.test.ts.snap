// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/index.vue { locale: en } snapshot test renders correctly 1`] = `
<section
  class="section"
>
  <div
    class="columns is-mobile"
  >
    <card-stub
      icon="github"
      title="Free"
    >
      <span>
        Open source on 
        <a
          href="https://github.com/oruga-ui/oruga"
        >
          GitHub
        </a>
      </span>
    </card-stub>
     
    <card-stub
      icon="cellphone-link"
      title="Responsive"
    >
      <span>
        <b
          class="has-text-grey"
        >
          Every
        </b>
         component is responsive
      </span>
    </card-stub>
     
    <card-stub
      icon="alert-decagram"
      title="Modern"
    >
      <span>
        Built with 
        <a
          href="https://vuejs.org/"
        >
          Vue.js
        </a>
         and 
        <a
          href="http://bulma.io/"
        >
          Bulma
        </a>
      </span>
    </card-stub>
     
    <card-stub
      icon="arrange-bring-to-front"
      title="Lightweight"
    >
      
      No other internal dependency
    
    </card-stub>
     
    <counter-stub />
  </div>
</section>
`;

exports[`pages/index.vue { locale: ja } snapshot test renders correctly 1`] = `
<section
  class="section"
>
  <div
    class="columns is-mobile"
  >
    <card-stub
      icon="github"
      title="無料"
    >
      <span>
        <a
          href="https://github.com/oruga-ui/oruga"
        >
          GitHub
        </a>
         上でオープンソース開発
      </span>
    </card-stub>
     
    <card-stub
      icon="cellphone-link"
      title="レスポンシブ デザイン"
    >
      <span>
        <b
          class="has-text-grey"
        >
          すべての
        </b>
        コンポーネントがレスポンシブ
      </span>
    </card-stub>
     
    <card-stub
      icon="alert-decagram"
      title="モダンUI"
    >
      <span>
        <a
          href="https://vuejs.org/"
        >
          Vue.js
        </a>
         と 
        <a
          href="http://bulma.io/"
        >
          Bulma
        </a>
         で作成
      </span>
    </card-stub>
     
    <card-stub
      icon="arrange-bring-to-front"
      title="軽量"
    >
      
      他のライブラリに依存しない
    
    </card-stub>
     
    <counter-stub />
  </div>
</section>
`;
